be/src/exec/json_parser.cpp

```diff
@@ -40,6 +40,56 @@ Status JsonDocumentStreamParser::parse(char* data, size_t len, size_t allocated)
     return Status::OK();
 }
 
+Status JsonDocumentStreamParser::_get_current_impl(simdjson::ondemand::object* row) {
+    while (true) {
+        try {
+            if (_doc_stream_itr != _doc_stream.end()) {
+                simdjson::ondemand::document_reference doc = *_doc_stream_itr;
+                // simdjson version 3.9.4 and JsonFunctions::to_json_string may crash when json is invalid.
+                // https://github.com/StarRocks/StarRocksTest/issues/8327
+                // TODO: add value in error message
+                if (doc.type() == simdjson::ondemand::json_type::array) {
+                    return Status::DataQualityError(
+                            "The value is array type in json document stream, you can set strip_outer_array=true to parse "
+                            "each element of the array as individual rows");
+                } else if (doc.type() != simdjson::ondemand::json_type::object) {
+                    return Status::DataQualityError("The value should be object type in json document stream");
+                }
+
+                _curr = doc.get_object();
+                *row = _curr;
+                _curr_ready = true;
+
+                _last_begin_offsets = _doc_stream_itr.current_index();
+
+                if (!_first_object_parsed) {
+                    _first_object_parsed = true;
+                }
+
+                return Status::OK();
+            }
+            return Status::EndOfFile("all documents of the stream are iterated");
+        } catch (simdjson::simdjson_error& e) {
+            if (e.error() != simdjson::CAPACITY || _batch_size == _len) {
+                // continue increasing the batch size is useless
+                throw;
+            }
+
+            _batch_size = std::min(_batch_size * 2, _len);
```

Comment:
this batch_size can be changed again and again, should forbid the parser from being used to parse multiple docs, the _batch_size determined by the previous document may not be suitable for next doc.

or you have the way to reset the batch_size before parsing the next doc.

Reply:
updated